src/integrations/prefect-gcp/prefect_gcp/cloud_storage.py

```diff
@@ -739,17 +739,38 @@ async def put_directory(
                 remote_file_path = str(
                     PurePosixPath(to_path, local_file_path.relative_to(local_path))
                 )
+                files_to_upload.append((local_file_path, remote_file_path))
+
+        # Upload files concurrently, reusing a single client
+        if files_to_upload:
+            # Get a single client and bucket object to reuse
+            client = self.gcp_credentials.get_cloud_storage_client()
+            bucket_obj = await run_sync_in_worker_thread(client.get_bucket, self.bucket)
+
+            upload_tasks = []
+            for local_file_path, remote_file_path in files_to_upload:
                 local_file_content = local_file_path.read_bytes()
-                with disable_run_logger():
-                    await cloud_storage_upload_blob_from_string.fn(
-                        data=local_file_content,
-                        bucket=self.bucket,
-                        blob=remote_file_path,
-                        gcp_credentials=self.gcp_credentials,
-                    )
-                uploaded_file_count += 1
-
-        return uploaded_file_count
+                blob_obj = bucket_obj.blob(remote_file_path)
+
+                upload_task = run_sync_in_worker_thread(
+                    blob_obj.upload_from_string, local_file_content
+                )
+                upload_tasks.append(upload_task)
+
+            # Execute uploads concurrently with a semaphore to limit parallelism
+            # Use 8 workers by default to balance speed and resource usage
+            semaphore = asyncio.Semaphore(8)
+
+            async def upload_with_semaphore(task):
+                async with semaphore:
+                    with disable_run_logger():
```

Comment:
You don't need `disable_run_logger` anymore because you aren't using `cloud_storage_upload_blob_from_string`, right?
```suggestion
```